src/components/TodoItem.tsx

```diff
@@ -0,0 +1,139 @@
+import React, {
+  FormEvent,
+  useState,
+  KeyboardEvent,
+  useRef,
+  useEffect,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  loadingTodosIds: number[];
+  onDelete?: (todoId: number) => Promise<void>;
+  onUpdate?: (todoToUpdate: Todo) => Promise<void>;
+  onChangeCompleted?: (todo: Todo) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  loadingTodosIds,
+  onDelete = () => {},
+  onUpdate = () => {},
+  onChangeCompleted = () => {},
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isEditing) {
+      inputRef.current?.focus();
+    }
+  }, [isEditing]);
```

Comment:
Move the `useEffect `to the end of code
Placing `useEffect `at the end improves readability by letting developers first grasp the main logic before seeing side effects. This promotes clearer structure and consistency in the codebase.